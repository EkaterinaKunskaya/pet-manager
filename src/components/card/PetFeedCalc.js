import React, { useState, useRef } from 'react';
import { countFeed } from '../helpers/СountFeed';

import '../../styles/PetFeedCalc.css';

const PetFeedCalc = ({ card }) => {

    const [energy, setEnergy] = useState(0);
    const [foodSpoons, setFoodSpoons] = useState(0);
    const [dailyAmount, setDailyAmount] = useState(0);

    let caloriesRef = useRef();
    let activityRef = useRef();
    let weightRef = useRef();

    function count() {
        const result = countFeed(caloriesRef.current.value, activityRef.current.value, weightRef.current.value, card.view)
        setDailyAmount(result.result);
        setFoodSpoons(result.resultSpoons);
        setEnergy(result.resultEnergy);
    }

    function clearCalcForm() {
        setEnergy(0);
        setFoodSpoons(0);
        setDailyAmount(0);
        caloriesRef.current.value = '';
        activityRef.current.value = '';
        weightRef.current.value = '';
    }

    return (
        <div className='PetFeedCalc'>
            <div className="calc-block">
                <h1>{`Калькулятор суточной нормы сухого корма для ${(card.view === 'Кошка') ? 'кошек' : 'собак'}`}</h1>
                <p>
                    Зная калорийность сухого корма, Вы можете рассчитать,
                    сколько корма потребуется в день Вашему питомцу.
                </p>

                <div className="calc-inputs">
                    <div>
                        <label htmlFor="calories">1. Введите калорийность корма, ккал / 100 г (энергетическую ценность):   </label>
                        <input type="text" id="calories" ref={caloriesRef} required />
                    </div>
                    <div>
                        <label htmlFor="activity">2. Выберите уровень активности:   </label>
                        <select id="activity" ref={activityRef} required>
                            <option value=""></option>
                            {(card.view === 'Кошка') ?
                                <>
                                    <option value="265">Домашняя</option>
                                    <option value="418">Активная</option>
                                    <option value="585">Беременная</option>
                                    <option value="419">Кормящая</option>
                                </>
                                : <>
                                    <option value="377">Низкая</option>
                                    <option value="460">Нормальная</option>
                                    <option value="523">Повышенная</option>
                                    <option value="680">Высокая</option>
                                    <option value="730">Щенки</option>
                                </>
                            }
                        </select>
                    </div>
                    <div>
                        <label htmlFor="weight">{`3. Введите вес ${(card.view === 'Кошка') ? 'кошки' : 'собаки'}, кг:   `}</label>
                        <input type="text" id="weight" ref={weightRef} required />
                    </div>
                </div>

                <div className='calc-btns'>
                    <button className="calc-btn" data-testid='feed-count' onClick={count}>Рассчитать</button>
                    <button className="calc-btn" data-testid='feed-clear' onClick={clearCalcForm}>Очистить</button>
                </div>

                <div className="calc-results">
                    <div className="calc-result">
                        <span><b>{`Рекомендуемая суточная норма сухого корма, г:   ${dailyAmount}`}</b></span>
                    </div>
                    <div className="calc-result">
                        <span><b>{`Количество сухого корма в столовых ложках:   ${foodSpoons}`}</b></span>
                        <p>* Не является точным значением. Зависит от величины и веса гранул сухого корма.</p>
                    </div>
                    <div className="calc-result">
                        <span><b>{`Энергетическая ценность корма, кДж / кг:   ${energy}`}</b></span>
                    </div>
                </div>

                <div className="calc-copy">
                    <p>
                        Норма потребления рассчитана по методике FEDIAF <br />
                        ("Федерация производителей кормов для домашних животных").
                    </p>
                </div>

            </div>
        </div>
    );

};

export default React.memo(PetFeedCalc);