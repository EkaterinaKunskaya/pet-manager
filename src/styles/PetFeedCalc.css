.PetFeedCalc {
    background: white;
    margin-top: -16px;
    border-radius: 0 8px 8px 8px;
    padding: 30px 50px;
}

.PetFeedCalc .calc-block {
    width: 90%;
    margin: 0 auto;
}

.PetFeedCalc .calc-block h1 {
    font-family: 'Pacifico', sans-serif;
    font-size: .9rem;
    text-align: center;
}

.PetFeedCalc .calc-block p {
    font-size: .5rem;
    text-align: center;
    font-family: sans-serif;
}

.PetFeedCalc .calc-inputs {
    margin-top: 30px;
    font-size: .6rem;
    font-family: sans-serif;
    white-space: pre;
}

.PetFeedCalc .calc-inputs input,
.PetFeedCalc .calc-inputs select {
    width: 30%;
    border: 1px solid rgba(0, 0, 0, 0.609);
    border-radius: 8px;
    padding: 5px;
    margin-bottom: 10px;
}

.PetFeedCalc .calc-btns {
    width: 60%;
    margin: 0 auto;
    display: flex;
    justify-content: space-between;
    padding: 20px 0;
}

.PetFeedCalc .calc-btn {
    width: 130px;
    height: 40px;
    border-radius: 45px;
    font-family: 'Pacifico', sans-serif;
    font-size: .6rem;
    text-align: center;
    font-weight: 400;
    color: black;
    background: rgba(255, 215, 0, 0);
    border: 1px solid rgba(0, 0, 0, 0.609);
    box-shadow: 0 8px 15px rgba(0, 0, 0, .2);
    transition: .3s;
}

.PetFeedCalc .calc-btn:hover {
    background: #00bdfb;
    box-shadow: 0 15px 20px rgba(46, 229, 157, .4);
    color: white;
    transform: translateY(-7px);
    border: 1px solid #00bdfb;
    cursor: pointer;
}

.PetFeedCalc .calc-results {
    margin-top: 10px;
    font-family: sans-serif;
    font-size: .6rem;
}

.PetFeedCalc .calc-result {
    white-space: pre;
    padding-bottom: 10px;
}

.PetFeedCalc .calc-result p {
    text-align: left;
    font-size: .5rem;
}

.PetFeedCalc .calc-copy p {
    margin-top: 20px;
    font-family: sans-serif;
    font-size: .5rem;
}


@media (max-width: 420px) {
    
    .PetFeedCalc {
        width: 100%;
        margin-top: -16px;
        border-radius: 0 8px 8px 8px;
        padding: 6% 0;
    }

    .PetFeedCalc .calc-block {
        width: 90%;
        margin: 0 auto;
    }

    .PetFeedCalc .calc-block h1 {
        font-size: 1.1rem;
        text-align: center;
    }

    .PetFeedCalc .calc-block p {
        display: none;
    }

    .PetFeedCalc .calc-inputs {
        width: 100%;
        margin-top: 20px;
        font-size: .9rem;
        white-space: normal;
    }

    .PetFeedCalc .calc-item {
        width: 200px;
    }

    .PetFeedCalc .calc-inputs label {
        word-wrap: break-word;
    }

    .PetFeedCalc .calc-inputs input,
    .PetFeedCalc .calc-inputs select {
        width: 100%;
        border-radius: 2px;
        padding: 5px;
        margin-bottom: 10px;
    }

    .PetFeedCalc .calc-btns {
        width: 100%;
        margin: 0 auto;
        display: block;
        padding: 20px 0;
    }

    .PetFeedCalc .calc-btn {
        width: 250px;
        height: 40px;
        border-radius: 8px;
        font-size: 1rem;
        margin-bottom: 15px;
    }

    .PetFeedCalc .calc-results {
        margin-top: 10px;
        font-family: sans-serif;
        font-size: 1.3rem;
    }

    .PetFeedCalc .calc-result {
        white-space: normal;
        padding-bottom: 10px;
    }

    .PetFeedCalc .calc-result p {
        text-align: left;
        font-size: 1rem;
    }

    .PetFeedCalc .calc-copy p {
        display: none;
    }
}