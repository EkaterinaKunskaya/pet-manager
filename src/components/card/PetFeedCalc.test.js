import React from 'react';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

import { PetFeedCalc } from './PetFeedCalc';

test('работа PetFeedCalc', () => {

  // создаём тестовую версию компонента
  const component = renderer.create(
    <PetFeedCalc card={{id:"ZTBaQrUWvPV0kMKsnduwL", petName:"Милка",view:"Собака"}} />
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента поле ввода и "введём" туда "5"
  // const inputElem = component.root.findByProps({"id":"calories"}); 
  // const inputElem = component.root.findByProps({"id":"activity"}); 
  // const inputElem = component.root.findByProps({"id":"weight"}); 
  // любые действия, вызывающие изменение стейта тестируемого
  // компонента, надо оборачивать в хелпер act

  // const countBtn = component.root.findByProps({"data-testid":"feed-count"}); 

  // renderer.act(() => {
  //   // делаем вид, что произошло событие onChange
  //   // сработает обработчик, поменяет стейт, а от него поменяется цифра в поле
  //   // const mockChangeEvent = { target: { value: '250' } };
  //   countBtn.props.onClick();
  // });

  // и получаем изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();



  
  
  
  //   // manually trigger the callback
  //   renderer.act(() => {
  //     tree.props.onMouseEnter();
  //   });
  //   // re-rendering
  //   tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  
  //   // manually trigger the callback
  //   renderer.act(() => {
  //     tree.props.onMouseLeave();
  //   });
  //   // re-rendering
  //   tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
  
  



  
});
