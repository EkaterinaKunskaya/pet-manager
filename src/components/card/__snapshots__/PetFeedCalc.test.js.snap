// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа PetFeedCalc 1`] = `
<div
  className="PetFeedCalc"
>
  <div
    className="calc-block"
  >
    <h1>
      Калькулятор суточной нормы сухого корма для собак
    </h1>
    <p>
      Зная калорийность сухого корма, Вы можете рассчитать, сколько корма потребуется в день Вашему питомцу.
    </p>
    <div
      className="calc-inputs"
    >
      <div>
        <label
          htmlFor="calories"
        >
          1. Введите калорийность корма, ккал / 100 г (энергетическую ценность):   
        </label>
        <input
          id="calories"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="activity"
        >
          2. Выберите уровень активности:   
        </label>
        <select
          id="activity"
          required={true}
        >
          <option
            value=""
          />
          <option
            value="377"
          >
            Низкая
          </option>
          <option
            value="460"
          >
            Нормальная
          </option>
          <option
            value="523"
          >
            Повышенная
          </option>
          <option
            value="680"
          >
            Высокая
          </option>
          <option
            value="730"
          >
            Щенки
          </option>
        </select>
      </div>
      <div>
        <label
          htmlFor="weight"
        >
          3. Введите вес собаки, кг:   
        </label>
        <input
          id="weight"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="calc-btns"
    >
      <button
        className="calc-btn"
        data-testid="feed-count"
        onClick={[Function]}
      >
        Рассчитать
      </button>
      <button
        className="calc-btn"
        onClick={[Function]}
      >
        Очистить
      </button>
    </div>
    <div
      className="calc-results"
    >
      <div
        className="calc-result"
      >
        <span>
          <b>
            Рекомендуемая суточная норма сухого корма, г:   0
          </b>
        </span>
      </div>
      <div
        className="calc-result"
      >
        <span>
          <b>
            Количество сухого корма в столовых ложках:   0
          </b>
        </span>
        <p>
          * Не является точным значением. Зависит от величины и веса гранул сухого корма.
        </p>
      </div>
      <div
        className="calc-result"
      >
        <span>
          <b>
            Энергетическая ценность корма, кДж / кг:   0
          </b>
        </span>
      </div>
    </div>
    <div
      className="calc-copy"
    >
      <p>
        Норма потребления рассчитана по методике FEDIAF 
        <br />
        ("Федерация производителей кормов для домашних животных").
      </p>
    </div>
  </div>
</div>
`;

exports[`работа PetFeedCalc 2`] = `
<div
  className="PetFeedCalc"
>
  <div
    className="calc-block"
  >
    <h1>
      Калькулятор суточной нормы сухого корма для собак
    </h1>
    <p>
      Зная калорийность сухого корма, Вы можете рассчитать, сколько корма потребуется в день Вашему питомцу.
    </p>
    <div
      className="calc-inputs"
    >
      <div>
        <label
          htmlFor="calories"
        >
          1. Введите калорийность корма, ккал / 100 г (энергетическую ценность):   
        </label>
        <input
          id="calories"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="activity"
        >
          2. Выберите уровень активности:   
        </label>
        <select
          id="activity"
          required={true}
        >
          <option
            value=""
          />
          <option
            value="377"
          >
            Низкая
          </option>
          <option
            value="460"
          >
            Нормальная
          </option>
          <option
            value="523"
          >
            Повышенная
          </option>
          <option
            value="680"
          >
            Высокая
          </option>
          <option
            value="730"
          >
            Щенки
          </option>
        </select>
      </div>
      <div>
        <label
          htmlFor="weight"
        >
          3. Введите вес собаки, кг:   
        </label>
        <input
          id="weight"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="calc-btns"
    >
      <button
        className="calc-btn"
        data-testid="feed-count"
        onClick={[Function]}
      >
        Рассчитать
      </button>
      <button
        className="calc-btn"
        onClick={[Function]}
      >
        Очистить
      </button>
    </div>
    <div
      className="calc-results"
    >
      <div
        className="calc-result"
      >
        <span>
          <b>
            Рекомендуемая суточная норма сухого корма, г:   0
          </b>
        </span>
      </div>
      <div
        className="calc-result"
      >
        <span>
          <b>
            Количество сухого корма в столовых ложках:   0
          </b>
        </span>
        <p>
          * Не является точным значением. Зависит от величины и веса гранул сухого корма.
        </p>
      </div>
      <div
        className="calc-result"
      >
        <span>
          <b>
            Энергетическая ценность корма, кДж / кг:   0
          </b>
        </span>
      </div>
    </div>
    <div
      className="calc-copy"
    >
      <p>
        Норма потребления рассчитана по методике FEDIAF 
        <br />
        ("Федерация производителей кормов для домашних животных").
      </p>
    </div>
  </div>
</div>
`;
